src/NexusMods.App.UI/Controls/TreeDataGrid/ColumnContentControl.cs

```diff
@@ -42,76 +43,29 @@ public DataTemplate DataTemplate
     }
 }
 
-/// <summary>
-/// Control for columns where row models are <see cref="CompositeItemModel{TKey}"/>.
-/// </summary>
-public class ColumnContentControl<TKey> : ContentControl
-    where TKey : notnull
+public abstract class ACustomContentControl<TContent> : ContentControl
+    where TContent : class
 {
-    [SuppressMessage("ReSharper", "CollectionNeverUpdated.Global", Justification = "Updated in XAML")]
-    public List<IComponentTemplate> AvailableTemplates { get; } = [];
-
-    public Control? Fallback { get; set; }
-
     private readonly SerialDisposable _serialDisposable = new();
 
-    /// <summary>
-    /// Builds a control from the first template in <see cref="AvailableTemplates"/>
-    /// that matches with a component in the item model.
-    /// </summary>
-    private Control? BuildContent(CompositeItemModel<TKey> itemModel)
-    {
-        foreach (var template in AvailableTemplates)
-        {
-            if (!itemModel.TryGet(template.ComponentKey, template.ComponentType, out var component)) continue;
-
-            // NOTE(erri120): DataTemplate.Build doesn't use the
-            // data you give it, need to manually set the DataContext.
-            // Otherwise. the new control will inherit the parent context,
-            // which is CompositeItemModel<TKey>.
-            var control = template.DataTemplate.Build(data: null);
-            if (control is null) return null;
-
-            control.DataContext = component;
-            return control;
-        }
-
-        return Fallback;
-    }
-
-    /// <summary>
-    /// Subscribes to component changes in the item model and rebuilds the content.
-    /// </summary>
-    private IDisposable Subscribe(CompositeItemModel<TKey> itemModel)
-    {
-        return itemModel.Components
-            .ObserveChanged()
-            .ObserveOnUIThreadDispatcher()
-            .Subscribe((this, itemModel), static (_, state) =>
-            {
-                var (self, itemModel) = state;
-                if (self.Presenter is null) return;
-
-                var content = self.BuildContent(itemModel);
-                self.Presenter.Content = content;
-            });
-    }
+    protected abstract Control? BuildContentControl(TContent content);
+    protected abstract IDisposable Subscribe(TContent content);
```

Comment:
Could you add some comments explaining the purpose of the class and the `BuildContentControl` and `Subscribe` methods?